src/strategies/FluentStrategy.ts

```diff
@@ -0,0 +1,104 @@
+  hasInput: <K extends string>(arbitraryName: K) => boolean
+  getInput: <K extends string, A>(arbitraryName: K) => FluentPick<A>
+  handleResult: () => void
+}
+
+export class FluentStrategy implements FluentStrategyInterface {
+
+  /**
+   * Record of all the arbitraries used for composing a given test case.
+   */
+  public arbitraries: StrategyArbitraries = {}
+
+  /**
+   * Default constructor. Receives the FluentCheck configuration, which is used for test case generation purposes.
+   */
+  constructor(public readonly configuration: FluentConfig = {sampleSize: 1000, shrinkSize: 500}) {}
+
+  /**
+   * Allows the configuration of a given strategy
+   */
+  config(config: FluentConfig) {
+    for (const k in config) this.configuration[k] = config[k]
+    return this
+  }
+
+  /**
+   * Adds an arbitrary to the arbitraries record
+   */
+  addArbitrary<K extends string, A>(arbitraryName: K, a: Arbitrary<A>) {
+    this.arbitraries[arbitraryName] = {arbitrary: a, pickNum: 0, collection: []}
+    this.setArbitraryCache(arbitraryName)
+  }
+
+  /**
+   * Configures the information relative a specific arbitrary.
+   */
+  configArbitrary<K extends string>(arbitraryName: K, partial: FluentResult | undefined, depth: number) {
+    this.arbitraries[arbitraryName].pickNum = 0
+    this.arbitraries[arbitraryName].collection = []
+
+    if (depth === 0)
+      this.arbitraries[arbitraryName].collection = this.arbitraries[arbitraryName].cache !== undefined ?
+        this.arbitraries[arbitraryName].cache :
+        this.buildArbitraryCollection(this.arbitraries[arbitraryName].arbitrary)
+    else if (partial !== undefined)
+      this.shrink(arbitraryName, partial, depth)
+  }
+
+  /**
+   * Determines whether uniqueness should be taken into account while generating samples.
+   */
+  isDedupable() {
+    return false
+  }
+
+  /**
+   * Hook that acts as point of extension of the addArbitrary function and that enables the strategy to be cached.
+   */
+  setArbitraryCache<K extends string, A>(_arbitraryName: K) {}
+
+  /**
+   * Generates a once a collection of inputs for a given arbitrary
+   */
+  buildArbitraryCollection<A>(arbitrary: Arbitrary<A>, sampleSize = this.configuration.sampleSize): FluentPick<A>[] {
+    return this.isDedupable() ? arbitrary.sampleUnique(sampleSize) : arbitrary.sample(sampleSize)
+  }
+
+  /**
+   * Hook that acts as point of extension of the configArbitrary function and that enables an arbitrary to be shrinked.
+   */
+  shrink<K extends string>(_name: K, _partial: FluentResult | undefined, _depth: number) {}
+
+  /**
+   * Determines whether there are more inputs to be used for test case generation purposes. This function can use
+   * several factors (e.g. input size, time) to determine whether the generation process should be stoped or not.
+   *
+   * Returns true if there are still more inputs to be used; otherwise it returns false.
+   */
+  hasInput<K extends string>(arbitraryName: K): boolean {
```

Comment:
I believe this exists because this class should not be instantiated directly, right? Can't we tag it as `abstract` instead of having these `throws`?

Reply:
Typescript mixins do not work with abstract classes.